### ISSUE TITLE
Problems with `:first-of-type` `:nth-of-type` `last-of-type` and a hacky temporary solution.

### ISSUE THREAD

**Issue by jayantasamaddar** [emotion-js/emotion#2922]
**Current behavior:**

## Problems with `:first-of-type` `:nth-of-type` `last-of-type`.

The first such selector is ignored. The hacky solution I'm using is adding an extra selector above all of them to ensure this is the one that gets skipped instead.

### Setup

I want to create a ButtonGroup component that comprises multiple Buttons grouped together. When grouped together, I want to `unset` the top and bottom `border-radius` of the leftmost and rightmost buttons and all of the `border-radius` for the rest of the buttons. Buttons inherit the `border-radius` style from the parent component.

Example:

![ButtonGroup-Example](https://user-images.githubusercontent.com/33408788/195603475-4c9feeac-3c78-45ab-a168-b5868bcfc0a9.png)

This is a basic structure of a Story in Storybook that creates the above.

```tsx
const SegmentedTemplate: ComponentStory<typeof ButtonGroup> = (args) => (
  <ButtonGroup {...args}>
    <Button plain outline>
      Bold
    </Button>
    <Button plain outline>
      Italic
    </Button>
    <Button plain outline>
      Underline
    </Button>
  </ButtonGroup>
);
```

The styling looks like this,

```tsx
export const ButtonGroup = styled(UrsaButtonGroup)(
  ({ segmented, fullWidth, connectedTop }) => `
    display: flex;
    flex-wrap: ${segmented ? 'nowrap' : 'wrap'};
    flex-grow: ${fullWidth ? '1' : '0'};
    min-width: ${fullWidth ? '100%' : 'auto'};
    gap: ${segmented ? '0' : '10px'};
    align-items: center;
    margin-left: ${segmented ? '0' : '2px'};
    margin-top: ${segmented ? '0' : '2px'}  

    ${
      segmented &&
      `.Ursa-ButtonItem:nth-of-type(n+1) // Unnecessary code used to trigger the skip. If not provided, the next selector gets skipped.
        > .Ursa-ButtonContainer
        > button {}
      
      .Ursa-ButtonItem:nth-of-type(n+2):nth-last-of-type(n+2)
        > .Ursa-ButtonContainer
        > button {
        border-radius: unset;
      }

      .Ursa-ButtonItem:last-of-type > .Ursa-ButtonContainer > button {
        margin-left: -1px;
        border-top-left-radius: unset;
        border-bottom-left-radius: unset;
      }

      .Ursa-ButtonItem:first-of-type > .Ursa-ButtonContainer > button {
        margin-right: -1px;
        border-top-right-radius: unset;
        border-bottom-right-radius: unset;
      }`
    }
  }
  `
);
```

- **`:nth-of-type(n+2):nth-last-of-type(n+2)`** - Selects all children except the first and last.
- **`:first-of-type`** - Selects the first child
- **`:last-of-type`** - Selects the last child

That's all I needed.

However to get around the issue I had to add this extra, **`:nth-of-type(n+1)`** since Emotion seems to be skipping the first declaration of the selector. 

**To reproduce:**

Basically any styled component that can be nested (having multiple children) should do. You can simply create three divs with a `border-radius: 4px` and nest them inside one parent container as shown above. And then use the css selectors above to check.

**Expected behavior:**

- The selector that gets ignored shouldn't be ignored. It's not really clean code to write an extra selector hoping it gets skipped to avoid a bug.

**Environment information:**

- `react` version: 18.0.2
- `@emotion/react` version: 11.10.4
- `@emotion/styled` version: 11.10.4
